bee-protocol/src/protocol/helpers.rs

```diff
@@ -118,19 +120,8 @@ impl Protocol {
     // Solidifier
 
     pub fn trigger_milestone_solidification(target_index: MilestoneIndex) {
-        if let Some(target_hash) = tangle().get_milestone_hash(target_index) {
-            if !tangle().is_solid_transaction(&target_hash) {
-                traversal::visit_parents_depth_first(
-                    tangle(),
-                    target_hash,
-                    |hash, _, metadata| {
-                        !metadata.flags.is_solid() && !Protocol::get().requested_transactions.contains_key(&hash)
-                    },
-                    |_, _, _| {},
-                    |_, _, _| {},
-                    |missing_hash| Protocol::request_transaction(*missing_hash, target_index),
-                );
-            }
-        }
+        Protocol::get()
+            .milestone_solidifier_worker
+            .unbounded_send(MilestoneSolidifierWorkerEvent(target_index));
```

Comment:
So only the milestone validator is triggering the milestone solidification ? If yes, we probably don't need this helpers and don't need to keep the channel, just pass it to the validator ?

Reply:
well sort of, for gossiped milestones we trigger it too when we solidified the last non-gossiped milestone.